packages/amplify-category-geo/src/provider-controllers/populate.ts

```diff
@@ -0,0 +1,97 @@
+  let uniqueIdentifier: string = 'id';
+  const identifierWalkthroughOptions = [
+    {name: 'No I will use the root level "id" field on Feature type. Auto-Assign if missing (this will UPDATE the GeoJSON file)', value: IdentifierOption.RootLevelID },
+    {name: 'Yes I want use one of the Geofence(Feature) properties as an identifier', value: IdentifierOption.CustomProperty}
+  ];
+  const identifierOption = await prompter.pick<'one', string>('Do you have an identifier field in the Geofence(Feature) properties?',identifierWalkthroughOptions) as IdentifierOption;
+  if (identifierOption === IdentifierOption.CustomProperty) {
+    uniqueIdentifier = await prompter.input('Provide the name of the property to use as a unique geofence identifier. Do not use Personal Identifiable Information such as email, username etc:')
+  }
+  //Validate the json file against schema
+  let geoJSONObj: FeatureCollection;
+  const validationSpinner = ora('Validating your GeoJSON file...');
+  validationSpinner.start();
+  try {
+    geoJSONObj = validateGeoJSONFile(geoJSONFilePath, uniqueIdentifier, identifierOption);
+    validationSpinner.succeed('Successfully validated GeoJSON file.');
+  } catch (err) {
+    validationSpinner.fail('Error occurs while validating GeoJSON file.');
+    throw err;
+  }
+  //Update the GeoJSON file with auto-assigned ID
+  if (identifierOption === IdentifierOption.RootLevelID) {
+    writeFileSync(geoJSONFilePath, JSON.stringify(geoJSONObj, null, 2));
+  }
+  //Construct geofence collection parameters
+  const geofenceCollectionParams = constructGeofenceCollectionParams({collectionToPopulate, uniqueIdentifier, identifierOption, geoJSONObj});
+  //Upload geofences to collection
+  const uploadSpinner = ora('Updating your Geofences in the collection...');
+  uploadSpinner.start();
+  try {
+    const result = await bulkUploadGeofence(geofenceCollectionParams, collectionRegion);
+    uploadSpinner.succeed(`Successfully added/updated ${result.Successes.length} Geofences in your "${collectionToPopulate}" collection`);
+  } catch (err) {
+    uploadSpinner.fail('Error occurs while uploading geofences.');
+    throw err;
+  }
+};
+
+const constructGeofenceCollectionParams = (populateParam: PopulateParams): GeofenceCollectionParams => {
+  const Entries: GeofenceParams[] = [];
+  const { geoJSONObj } = populateParam;
+  geoJSONObj.features.forEach(feature => {
+    Entries.push({
+      GeofenceId: populateParam.identifierOption === IdentifierOption.CustomProperty
+        ? feature.properties[populateParam.uniqueIdentifier]
+        : feature.id,
+      Geometry: {
+        Polygon: feature.geometry.coordinates
+      }
+    })
+  })
+  return {
+    CollectionName: populateParam.collectionToPopulate,
+    Entries
+  }
+}
+
+const bulkUploadGeofence = async (params: GeofenceCollectionParams, region: string) => {
+  const service = new Location({region});
+  return await service.batchPutGeofence(params).promise();
```

Comment:
Is this going to upload all the geofences in the file at once? There seems to be a limit of 10 geofences per batch https://docs.aws.amazon.com/location-geofences/latest/APIReference/API_BatchPutGeofence.html#API_BatchPutGeofence_RequestBody

Reply:
Added the limit for each batch